spec/lib/miq_pglogical/miq_pglogical_provider_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'miq_pglogical'
+
+describe MiqPglogicalProvider do
+  let(:connection) { ActiveRecord::Base.connection }
+  let(:pglogical)  { connection.pglogical }
+  subject          { described_class.new(connection) }
+
+  before do
+    skip "pglogical must be installed" unless pglogical.installed?
+    MiqServer.seed
+    subject.configure_provider
+  end
+
+  describe "#provider?" do
+    it "is true when a provider is configured" do
+      expect(subject.provider?).to be true
+    end
+  end
+
+  describe "#node?" do
+    it "is true when a provider is configured" do
+      expect(subject.node?).to be true
+    end
+  end
+
+  describe "#destroy_provider" do
+    it "removes the provider configuration" do
+      subject.destroy_provider
+      expect(pglogical.nodes.num_tuples).to eq(0)
+      expect(pglogical.replication_sets).not_to include(described_class::REPLICATION_SET_NAME)
+      expect(subject.provider?).to be false
+      expect(subject.node?).to be false
+    end
+  end
+
+  describe "#create_replication_set" do
+    it "creates the correct initial set" do
+      expected_excludes = subject.configured_excludes
+      actual_excludes = connection.tables - subject.included_tables
+      expect(actual_excludes).to match_array(expected_excludes)
+    end
+  end
+
+  describe "#refresh_excludes" do
+    it "adds a new non excluded table" do
+      connection.exec_query(<<-SQL)
+        CREATE TABLE test (id INTEGER PRIMARY KEY)
```

Comment:
The rspec transaction actually takes care of it this way.  Do we not want to just rely on that?